### PR TITLE
Jetpack Connect: Fix crash in Plans page after connecting a site with paid plan

### LINKED ISSUE #9209: Jetpack Connect: Crash in Plans page after connecting a site with paid plan
This issue might be related with the increasing Jetpack connection errors recently - cc @roccotripaldi.

#### Steps to reproduce
1. Go to the wp-admin of a disconnected Jetpack site that has a Premium plan.
2. Click the "Connect Jetpack" button in wp-admin to start authorization process. 
3. When the authorization window loads, click on the "Approve" button.

#### What I expected
To get redirected back to wp-admin (where the connection started) after connecting.

#### What happened instead
Window crashed with a "Maximum call stack size exceeded" error.

#### Browser / OS version
Google Chrome 54.0.2840.71 (64-bit) on Mac OS X El Capitan 10.11.6

#### Screenshot / Video
 
Provided kindly by @dereksmart (thanks).

#### Context / Source
Discovered and reported by @dereksmart.

### PR DESCRIPTION
**Pull request by tyxla**
### Purpose

This PR fixes #9209, where the window crashed after connecting a site with a premium plan when the connection process started from wp-admin. It introduces a new selector to check whether we're currently redirecting to wp-admin (includes tests).

The PR also addresses the edge case when the user visits `/jetpack/connect/plans/$site` with an active plan but without an active authorization session. In that case it was supposed to redirect to wp-admin, ut it didn't - redirection to wp-admin was failing because when the authorization `queryObject` was not there, so its `redirect_after_auth` was not defined. The PR resolves this by redirecting to wp-admin without using the authorization object in that case.

### To test:

**Test case 1**

* Disconnect one of your Jetpack sites that has a Premium plan.
* Go to /wp-admin/admin.php?page=jetpack of your Jetpack site.
* Click the "Connect Jetpack" button to start authorization.
* Add `&calypso_env=development` to the end of the long authorization URL.
* In the authorization window, click the "Approve" button.
* Verify you're redirected to wp-admin after connection process has completed, and there are no errors in the console in the meantime.

**Test case 2**

* Visit `/jetpack/connect/plans/$site`, where `$site` is a connected Jetpack sites that has a Premium plan.
* Verify you're properly redirected to My Plan page with no errors in the console.

Also, verify tests of the new `isRedirectingToWpAdmin()` selector are passing: `npm run test-client client/state/jetpack-connect/test/selectors.js -- --grep "isRedirectingToWpAdmin"`

/cc @dereksmart @roccotrip @johnHackworth @ryelle @oskosk

### CODE REVIEW

**Comment by lamosty** [comment]
Nice work @tyxla!

Re test case 1: how do we usually test this on local Calypso instance? After clicking on the "Connect Jetpack" button on my Jetpack testing site, I'm redirected to wp.com connect flow.

Re test case 2: after navigating to http://calypso.localhost:3000/jetpack/connect/plans/$site with a Premium connected Jetpack site, I'm redirected to `plans/my-plan/$site` instead of `wp-admin` of the Jetpack site. However, I get no console errors.

Tests are passing. :)

File: client/state/jetpack-connect/test/selectors.js [hunk 1]
```diff
@@ -354,6 +355,42 @@ describe( 'selectors', () => {
...
+					jetpackConnectAuthorize: {
+						isRedirectingToWpAdmin: false
+					}
+				}
+			};
+
+			expect( isRedirectingToWpAdmin( state ) ).to.be.false;
+		} );
+
+		it( 'should return false if redirection flag is set to true', () => {
```
Comment (lamosty) [inline client/state/jetpack-connect/test/selectors.js]: "should return `true`" instead of "should return `false`" :D

File: client/state/jetpack-connect/test/selectors.js [hunk 2]
```diff
@@ -354,6 +355,42 @@ describe( 'selectors', () => {
 		} );
 	} );
 
+	describe( '#isRedirectingToWpAdmin()', () => {
```
Comment (lamosty) [inline client/state/jetpack-connect/test/selectors.js]: I think you forgot to test what happens if the `state` does't include `jetpackConnect` and/or `jetpackConnectAuthorize` objects. It should probably return `null`.

Comment (tyxla) [inline client/state/jetpack-connect/test/selectors.js]: I don't think it's necessary - they will always be there.

**Comment by tyxla** [comment]
@lamosty thanks for the review :+1:

> how do we usually test this on local Calypso instance? After clicking on the "Connect Jetpack" button on my Jetpack testing site, I'm redirected to wp.com connect flow.

Good point, you should add `&calypso_env=development` to the end of the long authorization URL that you're redirected to after you click the "Connect Jetpack" button. I've added that as a separate step in the test instructions.

> I'm redirected to plans/my-plan/$site instead of wp-admin of the Jetpack site. 

Ah, that makes sense, it was introduced recently in another PR. Updating the test instructions.

Please, let me know if you have any other finds :)

**Comment by johnHackworth** [comment]
I think we can merge this... :shipit: !
